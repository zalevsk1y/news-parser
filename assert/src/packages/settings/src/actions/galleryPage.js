import {types} from './index';


export const addGallery=()=>{
    return{
        type:types.gallery.TOGGLE_ADD_GALLERY,
       
    }
}
export const shortCode=(text)=>{
    return{
        type:types.gallery.CHANGE_SHORTCODE,
        text:text
    }
}
export const parameterName=(text)=>{
    return{
        type:types.gallery.CHANGE_PARAMETER_NAME,
        text:text,
    }
}

