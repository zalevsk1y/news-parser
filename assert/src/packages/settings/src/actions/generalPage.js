import  {types} from './index';

export function toggleAddSource(){
    return{
        type:types.general.TOGGLE_ADDSOURCE
    }
}

