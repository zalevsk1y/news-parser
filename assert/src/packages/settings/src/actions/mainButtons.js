import {types} from './index';
import config from "@news-parser/config"
import {oldServerData,newServerData} from '@news-parser/helpers';   

//import {store} from '../index'
function sendToServer(){
    return {
        type:types.main.SEND_REQUEST,
    }
}
export function resetToDefault(){
    return {
        type:types.main.RESET_TO_DEFAULT
    }
}
function responseFromServer(data){
    return{
        type:types.main.RECEIVE_RESPONSE,
        data:data.main
    }
}
function setAllSettings(data){
    return {
        type:types.main.SET_ALL_SETTINGS,
        settings:data.settings
    }
}

export function saveSettingsToServer (dispatch,settings,nonce){
  
    const parameters=config.emulateJSON?oldServerData('settings',settings):newServerData(settings);
    return dispatch=>{
        dispatch(sendToServer())
        return fetch(config.settingsApi.saveSettings+'&_wpnonce='+nonce,
        parameters
       )
       .then(response=>response.json())
       .then((json,error)=>{
            
           if(!error&&json){
            dispatch(responseFromServer(json));
        
           }
       })
    }
}
export function getSettingsFromServer (dispatch,nonce){
    const parameters=config.emulateJSON?oldServerData():newServerData();
    return dispatch=>{ 
        return fetch(config.settingsApi.getSettings+'&_wpnonce='+nonce,parameters)
       .then(response=>response.json())
       .then((json,error)=>{
           if(!error&&json){
            dispatch(setAllSettings(json));
           }
       })
    }
}
export function getDefaultSettingsFromServer (dispatch,nonce){
    const parameters=config.emulateJSON?oldServerData():newServerData();
    return dispatch=>{ 
        return fetch(config.settingsApi.getDefaultSettings+'&_wpnonce='+nonce,parameters)
       .then(response=>response.json())
       .then((json,error)=>{
           if(!error&&json){
            dispatch(setAllSettings(json));
           }
       })
    }
}