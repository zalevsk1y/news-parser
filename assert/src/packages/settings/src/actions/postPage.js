import  {types} from './index';


export const addThumbnail=()=>{
    return{
        type:types.post.TOGGLE_ADD_THUMBNAIL,
      
    }
}
export const parseOtherPictures=()=>{
    return{
        type:types.post.TOGGLE_PARSE_OTHER_PICTURES,
    }
}
export const showPicturesDialog=()=>{
    return{
        type:types.post.TOGGLE_SHOW_PICTURES_DIALOG,
    }
}
export const maxPictures=(number)=>{
    return{
        type:types.post.CHANGE_MAX_PICTURES,
        number:number,
    }
}

