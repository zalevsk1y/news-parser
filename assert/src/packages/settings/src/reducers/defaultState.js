
export const defaultState={ 
    main:{
        isFetching:false,
        error:false,
        message:false,
    },
    settings:{
        general:{
            addSource:false
        },
        post:{
            addThumbnail:true,
            parseOtherPictures:false,
            showPicturesDialog:true,
            maxPictures:5,
        },
        gallery:{
            addGallery:false,
            shortCode:undefined,
            parameterName:undefined
        }
}
};
