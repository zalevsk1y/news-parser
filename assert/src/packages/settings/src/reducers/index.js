import general from './generalPage.js';
import post from './postPage.js';
import gallery from './galleryPage.js';
import {main} from './main.js'
import {combineReducers} from 'redux';

const settings=combineReducers({
    general,
    post,
    gallery,
})
const  settingsReducer=combineReducers({
    main,
    settings
})

function flowLeftToRight(funcObject){
    return (state,action)=>{
        let newState=state;
        const funcArray=Object.keys(funcObject)
        for(var i=0,end=funcArray.length;i<end;i++){
            var funcName=funcArray[i];
            newState=funcObject[funcName].apply(this,[newState,action]);
        }
        return newState;
    }
}
export default settingsReducer;