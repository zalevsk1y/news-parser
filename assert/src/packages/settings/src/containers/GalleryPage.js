import {connect} from 'react-redux';
import {galleryPage} from '../actions/index.js';
import React from 'react';
import Switch from '../components/Switch';
import Input from '../components/Input';
import {GALLERY_FIELD_NAME} from '../actions/index';
import PropTypes from 'prop-types';

export const GalleryPage=({settings,addGallery,shortCode,parameterName})=>(
            <div>
              <table className="settings-content">
                  <tbody>
                      <tr>
                          <th>
                              Add Gallery to post.
                          </th>
                          <td>
                              <Switch className="settings-switch add-gallery"  active={settings.addGallery}  onClick={addGallery} fieldName='addGallery'/>
                          </td>
                      </tr>
                      
                      </tbody>
              </table>         
                    <table className={"sub-settings-content "+(settings.addGallery?"visible":"hidden")}>
                        <tbody> 
                            <tr>
                                <th>
                                    Shortcode for gallery
                                </th>
                                <td>
                                    <Input className="settings-shortcode" onBlur={shortCode} fieldName='shortCode' value={settings.shortCode}/>
                                    <p className="description">
                                        You can add short code for additional gallery plugin.More <a href="#">info</a>. 
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <th>
                                    Parameter name for gallery
                                </th>
                                <td>
                                    <Input className="settings-parameter-name" onBlur={parameterName} fieldName='parameterName' value={settings.parameterName}/>
                                    <p className="description">
                                        You can specify name of parameter for pictures IDs.More <a href="#">info</a>. 
                                    </p>
                                </td>
                            </tr>
                        </tbody>
                    </table>
            </div>  
      );
    
const mapStateToProps = (state, ownProps) => {
        return {
            settings: state.settings[GALLERY_FIELD_NAME]
        }
    },

    mapDispatchToProps = (dispatch) => {
        return {
            addGallery: () => {
                dispatch(galleryPage.addGallery());
            },
            shortCode: (text) => {
                dispatch(galleryPage.shortCode(text));
            },
            parameterName: (text) => {
                dispatch(galleryPage.parameterName(text));
            }
        }
    }

    export default connect(mapStateToProps, mapDispatchToProps)(GalleryPage)

    GalleryPage.propTypes={
        settings:PropTypes.shape({
            addGallery:PropTypes.bool,
            shortCode:PropTypes.string,
            parameterName:PropTypes.string
        }).isRequired,
        addGallery:PropTypes.func.isRequired,
        shortCode:PropTypes.func.isRequired,
        parameterName:PropTypes.func.isRequired
    }