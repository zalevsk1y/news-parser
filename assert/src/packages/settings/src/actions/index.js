import {toggleAddSource} from './generalPage';
import {addThumbnail,parseOtherPictures,showPicturesDialog,maxPictures} from './postPage';
import {addGallery,shortCode,parameterName} from './galleryPage';
import {getDefaultSettingsFromServer,saveSettingsToServer,getSettingsFromServer} from './mainButtons';

export const types={
    main:{
        SEND_REQUEST:'SEND_REQUEST',
        RECEIVE_RESPONSE:'RECEIVE_RESPONSE',
        SET_ALL_SETTINGS:'SET_ALL_SETTINGS',
        RESET_TO_DEFAULT:'RESET_TO_DEFAULT'
    },
    general:{
        TOGGLE_ADDSOURCE:'TOGGLE_ADDSOURCE'
    },
    post:{
        TOGGLE_ADD_THUMBNAIL:'TOGGLE_DOWNLOAD_PICTURES',
        TOGGLE_SHOW_PICTURES_DIALOG:'TOGGLE_SHOW_PICTURES_DIALOG',
        CHANGE_MAX_PICTURES:'CHANGE_MAX_PICTURES',
        TOGGLE_PARSE_OTHER_PICTURES:'TOGGLE_PARSE_OTHER_PICTURES'
    },
    gallery:{
        TOGGLE_ADD_GALLERY:"TOGGLE_ADD_GALLERY",
        CHANGE_SHORTCODE:"CHANGE_SHORTCODE",
        CHANGE_PARAMETER_NAME:"CHANGE_PARAMETER_NAME"
    }

}


export const GALLERY_FIELD_NAME='gallery';
export const POST_FIELD_NAME='post';
export const GENERAL_FIELD_NAME='general';

export const  generalPage={toggleAddSource};
export const  postPage={addThumbnail,parseOtherPictures,showPicturesDialog,maxPictures};
export const  galleryPage={addGallery,shortCode,parameterName};
export const  main={getDefaultSettingsFromServer,saveSettingsToServer,getSettingsFromServer}
