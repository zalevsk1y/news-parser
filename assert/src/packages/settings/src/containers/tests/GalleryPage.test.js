import React from 'react';
import {shallow,configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {GalleryPage} from '../GalleryPage';
import {defaultState} from '../../reducers/defaultState'
import {GALLERY_FIELD_NAME} from '../../actions'

configure({adapter:new Adapter()});
describe('Gallery Page',()=>{
    it('should render correctly',()=>{
        const dispatchToProps={ 
                    addGallery:jest.fn(),
                    shortCode:jest.fn(),
                    parameterName:jest.fn()
            }
        const component=shallow(<GalleryPage settings={defaultState.settings[GALLERY_FIELD_NAME]} {...dispatchToProps}/>)
        expect(component).toMatchSnapshot();
    })
})