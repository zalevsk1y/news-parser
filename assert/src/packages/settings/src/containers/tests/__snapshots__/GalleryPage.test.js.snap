// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Page should render correctly 1`] = `
<div>
  <table
    className="settings-content"
  >
    <tbody>
      <tr>
        <th>
          <Connect(Translate)>
            Add gallery to post
          </Connect(Translate)>
        </th>
        <td>
          <Switch
            active={false}
            className="settings-switch add-gallery"
            fieldName="addGallery"
            onClick={[MockFunction]}
          />
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="sub-settings-content hidden"
  >
    <tbody>
      <tr>
        <th>
          <Connect(Translate)>
            Shortcode for gallery
          </Connect(Translate)>
        </th>
        <td>
          <Input
            className="settings-shortcode"
            fieldName="shortCode"
            onBlur={[MockFunction]}
          />
          <p
            className="description"
          >
            <Connect(Translate)>
              You can add short code for additional gallery plugin.More
            </Connect(Translate)>
              
            <a
              href="#"
            >
              <Connect(Translate)>
                info
              </Connect(Translate)>
            </a>
            .
          </p>
        </td>
      </tr>
      <tr>
        <th>
          <Connect(Translate)>
            Parameter name for gallery
          </Connect(Translate)>
        </th>
        <td>
          <Input
            className="settings-parameter-name"
            fieldName="parameterName"
            onBlur={[MockFunction]}
          />
          <p
            className="description"
          >
            <Connect(Translate)>
              You can specify name of parameter for pictures IDs.More
            </Connect(Translate)>
             
            <a
              href="#"
            >
              <Connect(Translate)>
                info
              </Connect(Translate)>
            </a>
            .
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
